opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -189,6 +231,35 @@ public function index() {
         $this->response->setOutput($this->render());
     }
 
+    protected function handleLayoutSaving() {
+        if (!$this->user->hasPermission('modify', 'module/smaily_for_opencart')) {
+            return;
+        }
+        $optin_form_settings = [];
+        // Load Smaily admin model for saving settings.
+        $this->load->model('smailyforopencart/admin');
+        // Check if layout settings are in POST & that it's an array.
+        if (!empty($this->request->post['smaily_optin_form_settings']) && is_array($this->request->post['smaily_optin_form_settings'])) {
+            // Declare layout module settings in POST.
+            $optin_form_settings = $this->request->post['smaily_optin_form_settings'];
+            // Loop over each array and validate array keys.
+            foreach ($smaily_modules as &$module) {
```

Comment:
You do not have variable `$smaily_modules` any more. You have changed it to `$optin_form_settings`. At the moment posted data is never sanitized or validated.